google-cloud-datastore/test/google/cloud/datastore/query_test.rb

```diff
@@ -69,6 +69,55 @@
     _(Google::Cloud::Datastore::Convert.from_value(second_filter.property_filter.value)).must_equal Time.new(2014, 1, 1, 0, 0, 0, 0)
   end
 
+  it "can filter properties through Filter objects" do
+    # Construct the filter (A OR B OR C) AND D
+    # ((a = 3 OR b > 4) OR c < 10) AND d > 11
+    filter = Google::Cloud::Datastore::Filter.new("a", "=", 3)
```

Comment:
Q: Does this mean that the order of filters is always left to write?

Reply:
That's correct. The chained method calls dictate the order of grouping.